patches/jsc.patch

```diff
@@ -291,3 +291,37 @@ diff -aur target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/S
  )
 +add_definitions(-DJSC_VERSION="${JSC_VERSION}")
 
+diff -aur target-org/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp target/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp
+--- target-org/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp	2018-08-24 16:08:22.000000000 +0800
++++ target/webkit/Source/WTF/wtf/linux/MemoryFootprintLinux.cpp	2018-11-07 22:59:05.000000000 +0800
+@@ -38,6 +38,30 @@
+ #if OS(LINUX)
+ static const Seconds s_memoryFootprintUpdateInterval = 1_s;
+
++#if defined(__ANDROID__) && __ANDROID_API__ < 18
++//
++// The POSIX.1 C getline isn't exported until API 18 in NDK.
++// As the getline() used to read content from /proc/self/smaps,
++// the content fortunately are plain text and lines are not too long.
++// Here to simply use fgets() as fallback.
++// Some differences further:
++//  1. Line buffer is fixed 512 bytes.
++//  2. Return 1 if succeed.
++//     This is only for forEachLine() use.
++//     Just don't want to do strlen() calculating length again.
++//
++static ssize_t getline(char** linep, size_t* linecapp, FILE* stream) {
```

Comment:
I checked libc binaries from android-16 NDK and found getline symbols there. So there is a chance that getline definition is only missing from the headers. I suggest (and we used similar trick in the past several times) to try providing only a declaration of getline function which then should be linked with the implementation present in libc:
```
extern "C" ssize_t getline(char** linep, size_t* linecapp, FILE* stream);
```

Reply:
@kmagiera This is awesome that I did not think before.
BTW how do you check getline existed in android-16 NDK?
I cannot find it on my machine 😓 

```shell
> pwd
android-ndk-r17c

> toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-readelf -Ws platforms/android-16/arch-arm/usr/lib/libc.so|grep getline

> toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/bin/arm-linux-androideabi-readelf -Ws platforms/android-18/arch-arm/usr/lib/libc.so|grep getline
    10: 000081b5     2 FUNC    GLOBAL DEFAULT    7 getline@@LIBC
  1241: 000081b5     2 FUNC    GLOBAL DEFAULT    7 getline
 
```
>

Reply:
I used `nm` instead of `readelf` but that shouldn't matter. Also, I only checked static version of libc (that is `libc.a`) but now I see the dynamic version doesn't have that symbol

Reply:
So sad. It would be great if getline is just a hidden symbol.